### PR TITLE
Android CHIP demo app

### PR DESCRIPTION
**Pull request by vidhis88**
Scan CHIP QR code and display the SetupPayload info.

Video: https://drive.google.com/file/d/1BqqjPsQItrdNCNtfjfR5W4zMKUVkiTpy/view

### CODE REVIEW

**Review by woody-apple — APPROVED** [review]

**Review by BroderickCarlin — APPROVED** [review]

**Comment by pan-apple** [comment]
Are the following files build artifacts being accidentally checked in?

```
src/android/CHIPTool/app/src/main/jniLibs/arm64-v8a/libcrypto.so
src/android/CHIPTool/app/src/main/jniLibs/arm64-v8a/libSetupPayloadParser.so
src/android/CHIPTool/app/src/main/jniLibs/x86/libSetupPayloadParser.so
src/android/CHIPTool/app/src/main/jniLibs/x86/libcrypto.so
```

File: src/android/CHIPTool/app/build.gradle [hunk 1]
```diff
@@ -15,7 +17,7 @@ android {
 
         // NOTE: This build assumes CHIP was configured and built for armeabi-v7a. Deal with
         // other archs later when build is sane.
-        ndk {abiFilters 'armeabi-v7a'}
+//        ndk {abiFilters 'armeabi-v7a'}
```
Comment (andy31415) [inline src/android/CHIPTool/app/build.gradle:20]: nit: should we align the comment slashes here?
```suggestion
        // ndk {abiFilters 'armeabi-v7a'}
```

**Review by andy31415 — COMMENTED** [review]
> Are the following files build artifacts being accidentally checked in?
> 
> ```
> src/android/CHIPTool/app/src/main/jniLibs/arm64-v8a/libcrypto.so
> src/android/CHIPTool/app/src/main/jniLibs/arm64-v8a/libSetupPayloadParser.so
> src/android/CHIPTool/app/src/main/jniLibs/x86/libSetupPayloadParser.so
> src/android/CHIPTool/app/src/main/jniLibs/x86/libcrypto.so
> ```

Maybe we could use some $NDK_PATH in our build scripts for the crypto bit. For the PayloadParser, it sounds as if our build should create that file and we should not need to check it in.

**Review by andy31415 — CHANGES_REQUESTED** [review]
Marking as changes requested: I believe we should not checkin .so files and we should instead require user paths if we cannot autodetect (e.g. chip build output directory or NDK path).

**Review by gerickson — CHANGES_REQUESTED** [review]
Looks good, but agreed we do not want to check in the .so binaries.

**Comment by randyrossi** [comment]
Same goes for .jar files.  Any artifacts of the build should not be checked in.

**Comment by woody-apple** [comment]
@vidhis88 @randyrossi any update here?

File: src/android/CHIPTool/app/build.gradle (hunk 1 above)
Comment (vidhis88) [reply src/android/CHIPTool/app/build.gradle]: Done.

**Comment by vidhis88** [comment]
Updated the PR to remove the .so and .jar files.

Since we don't yet have a build job for the Android app, I wanted to allow anyone checking out the Android demo app to be able to run it without having to manually build the .so and .jar files by including them.
But I do see the point of muddying git history and have updated the PR accordingly.

**Review by andy31415 — APPROVED** [review]

**Review by gerickson — APPROVED** [review]

**Pull request merged**